src/commando/core.py

```diff
@@ -1,2 +1,79 @@
-def hello_world():
-    return "Hello, world!"
+import subprocess
+import tempfile
+from functools import partial
+from subprocess import PIPE, STDOUT, CalledProcessError
+
+
+def _pprint(repo, message):
+    print(f"{repo}> {message}")
+
+
+def _run(cwd, results, command, **kwargs):
+    result = subprocess.run(  # pylint:disable=subprocess-run-check
+        command, cwd=cwd, stdout=PIPE, stderr=STDOUT, text=True, **kwargs
+    )
+    results.append(result)
+    result.check_returncode()
+    return result
+
+
+def _process_repo(
+    repo, command, commit_message, branch, pr_title, pr_body
+):  # pylint:disable=too-many-arguments
+
+    # Collect the results of all commands run for this repo.
+    results = []
+
+    pprint = partial(_pprint, repo)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        run = partial(_run, tmpdirname, results)
+
+        try:
+            pprint("cloning")
+            run(["gh", "repo", "clone", repo, tmpdirname, "--", "--quiet"])
+
+            pprint(command)
+            run(command, shell=True)
+
+            if run(["git", "status", "--porcelain"]).stdout:
```

Comment:
Might be able to simplify this and remove this `if`: command doesn't need to check whether or not there are changes. It can just go ahead blindly because gh-pr-upsert will do the right thing if there are no changes. I think the `git commit` command below will need `--allow-empty` to avoid crashing.